### PR TITLE
Add Fractional class

### PR DESCRIPTION
**Pull request by eliaslfox**

### CODE REVIEW

**Comment by stylewarning** [comment]
I don't actually see the definition of Fractional in the diff.

File: library/arith.lisp [hunk 1]
```diff
@@ -478,22 +479,39 @@ The fields are defined as follows:
     (define (fromInt z)
       (lisp Fraction (z) z))))
 
+(coalton-toplevel
```
Comment (eliaslfox) [inline library/arith.lisp:482]: @stylewarning  It's here

**Review by stylewarning — CHANGES_REQUESTED** [review]
I'm open to not making this change, but as a math-code writer, I prefer things to be more specific.

**Comment by eliaslfox** [comment]
Closing in favor of the type class hierarchy in #533

**Pull request closed without merging**